### PR TITLE
alter the edit link from the menu tree form

### PR DESCRIPTION
**Pull request by porchlight**

### CODE REVIEW

File: src/Plugin/Menu/MenuPositionLink.php [hunk 1]
```diff
@@ -16,9 +19,24 @@ class MenuPositionLink extends MenuLinkBase {
    * @param mixed $plugin_definition
    *   The plugin implementation definition.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityManagerInterface $entity_manager) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->entity_manager = $entity_manager;
```
Comment (star-szr) [inline src/Plugin/Menu/MenuPositionLink.php]: For coding standards the class property should be $this->entityManager. You also need to initialize that property on the class.

File: src/Form/MenuPositionRuleForm.php [hunk 2]
```diff
@@ -262,6 +262,7 @@ protected function getPluginDefinition($menu_name, $parent) {
     $definition['route_name'] = null;
     $definition['route_parameters'] = [];
     $definition['enabled'] = false;
+    $definition['metadata'] = array('entity_id' => $rule->getId());
```
Comment (star-szr) [inline src/Form/MenuPositionRuleForm.php:265]: This should be `menu_position_rid` or something instead of `entity_id`.

Comment (star-szr) [inline src/Form/MenuPositionRuleForm.php:265]: Nevermind, we know what menu link plugin it's using so we don't need it to be namespaced.

**Comment by midnightLuke** [comment]
This looks good to me, just need to run a quick test and I will approve it.

It should be noted that after this merge all old rules will be invalid (no metadata stored for them), so a reinstall of the module is probably necessary, unless we feel like we should be writing update hooks (or whatever they are in D8) at this point?

**Comment by star-szr** [comment]
IMO it's premature to worry about update hooks, we're in pre-alpha state right now so upgrade path shouldn't be expected.

**Comment by ckaotik** [comment]
Just for documentation, this PR relates to #19, let's have GitHub display it there :+1:

**Pull request merged**